parts/k8s/cloud-init/artifacts/cse_customcloud.sh

```diff
@@ -27,14 +27,54 @@ ensureCertificates() {
     fi
   fi
 
-  # ensureCertificates will be retried if the exit code is not 0
+  # ensureAzureStackCertificates will be retried if the exit code is not 0
   curl $AZURESTACK_RESOURCE_METADATA_ENDPOINT
   exit $?
 }
 
+ensureCustomCloudRootCertificates() {
```

Comment:
ditto above

Let's wrap this func in the go template layer conditional {{- if IsCustomCloudProfile}} (and then {{end}} at the end, so that we save cloud-init payload in Azure Stack scenarios.

Reply:
Actually it'll probably need to be `{{- if not IsAzureStackCloud}}` because `IsCustomCloudProfile` is inclusive of Azure Stack.

Reply:
To keep the code logic simple for now, I just used {{- if IsCustomCloudProfile }}. The moment we add this block, we are probably reducing code size for 95% of scenarios anyways. We can revisit in the future if we need to further reduce the code size for AzureStack scenario.